core/vtk/ttkMergeTreeDistanceMatrix/ttkMergeTreeDistanceMatrix.cpp

```diff
@@ -229,3 +309,113 @@ int ttkMergeTreeDistanceMatrix::RequestData(
 
   return res;
 }
+
+std::tuple<std::vector<float>, std::vector<std::vector<int>>, int>
+  ttkMergeTreeDistanceMatrix::ftmToAdjList(
```

Comment:
It would not be better to make this function returns void and put the actual output as an input parameter by reference to avoid copy?

Reply:
jeeze, good catch.
please do not return objects in general and prefer passing the output of your function by reference (as a parameter) as recalled by mathieu. returning outputs directly induces a copy, which can slow everything down when the object is large.

Reply:
Actually, compilers can use [Copy Elision](https://en.cppreference.com/w/cpp/language/copy_elision) to avoid copies in return statements (aka "Return Value Optimization").

Reply:
is that enabled by default on all compilers?

Reply:
I agree that this is not good style and I can fix this. However, this function is going to be removed anyway, as soon as I push the update to the FTM data structures.

Reply:
if you don't mind, I'd prefer that you fix that function right now (to make sure it does not stay too long in the source tree with a copy return). thanks!